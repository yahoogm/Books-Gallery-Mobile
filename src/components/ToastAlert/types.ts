export type ToastAlertProps = {
  id: string;
  message: string;
  action: 'success' | 'info' | 'error' | 'warning' | 'attention';
};
