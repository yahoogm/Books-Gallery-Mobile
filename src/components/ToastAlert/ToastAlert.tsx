import {VStack, Toast, ToastTitle} from '@gluestack-ui/themed';
import {ToastAlertProps} from './types';

const ToastAlert: React.FC<ToastAlertProps> = ({id, message, action}) => {
  return (
    <Toast nativeID={id} action={action} variant="solid">
      <VStack space="xs">
        <ToastTitle>{message}</ToastTitle>
      </VStack>
    </Toast>
  );
};

export default ToastAlert;
