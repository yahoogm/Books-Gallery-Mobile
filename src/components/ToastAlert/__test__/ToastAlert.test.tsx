import {render} from '@testing-library/react-native';
import ToastAlert from '../ToastAlert';

describe('Toast Alert', () => {
  it('match snapshot when given props', () => {
    const tree = render(
      <ToastAlert action="attention" id="1234" message="testing" />,
    );

    const instance = tree.toJSON();
    expect(instance).toMatchSnapshot();
  });
});
