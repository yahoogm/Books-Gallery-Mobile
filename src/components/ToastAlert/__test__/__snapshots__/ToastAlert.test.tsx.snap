// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Alert match snapshot when given props 1`] = `
<View
  action="attention"
  data-style=""
  dataSet={
    {
      "style": "",
    }
  }
  nativeID="1234"
  style={
    [
      undefined,
      {},
      undefined,
    ]
  }
  variant="solid"
>
  <View
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    space="xs"
    style={
      [
        undefined,
        {},
        undefined,
      ]
    }
  >
    <Text
      aria-atomic="true"
      aria-live="assertive"
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      role="alert"
      style={
        [
          undefined,
          {},
          undefined,
          {},
          undefined,
        ]
      }
    >
      testing
    </Text>
  </View>
</View>
`;
